File: official/docs/node/current/billing/add-stripe-bank-account.js
const EasyPostClient = require('@easypost/api');

const client = new EasyPostClient('EASYPOST_API_KEY');

(async () => {
  const paymentMethod = await client.ReferralCustomer.addBankAccountFromStripe(
    'REFERRAL_USER_API_KEY',
    'fca_...',
    {
      ip_address: '127.0.0.1',
      user_agent: 'Mozilla/5.0',
      accepted_at: 172251073,
    },
    'primary',
  );

  console.log(paymentMethod);
})();

File: official/docs/java/current/billing/add-stripe-bank-account.java
package billing;

import com.easypost.exception.EasyPostException;
import com.easypost.model.PaymentMethod;
import com.easypost.service.EasyPostClient;

public class CreateStripeCreditCardSecret {
    public static void main(String[] args) throws EasyPostException {
        EasyPostClient client = new EasyPostClient("EASYPOST_API_KEY");

        HashMap<String, Object> mandateData = new HashMap<String, Object>();
        mandateData.put("ip_address", "127.0.0.1");
        mandateData.put("user_agent", "Mozilla/5.0");
        mandateData.put("accepted_at", 1722510730);

        PaymentMethodObject paymentMethod = client.referralCustomer.addBankAccountFromStripe(
                "REFERRAL_USER_API_KEY",
                "fca_...",
                mandateData,
                PaymentMethod.Priority.PRIMARY
        );

        System.out.println(paymentMethod);
    }
}
